astroid/scoped_nodes.py

```diff
@@ -103,6 +103,39 @@ def _c3_merge(sequences, cls, context):
     return None
 
 
+def clean_typing_generic_mro(sequences: List[List["ClassDef"]]) -> None:
+    """typing.Generic is allowed to appear multiple times in the initial mro.
+    The final one however, MUST only contain ONE.
+
+    This method will check if Generic is in inferred_bases, but also
+    part of bases_mro. If true, remove it from inferred_bases
+    as well as its entry the bases_mro.
+
+    Format sequences: [[self]] + bases_mro + [inferred_bases]
+    """
+    pos_generic_in_main_bases = -1
+    # Check if part of inferred_bases
+    for i, base in enumerate(sequences[-1]):
```

Comment:
I think it would be definitely clearer to manipulate self (or an alias), bases_mro and inferred bases instead of dealing with only one sequence and multiples indexes.

Reply:
I would like to leave `list(clean_duplicates_mro(unmerged_mro, self, context))` in `_compute_mro` untouched. This should be run before the MRO is cleaned from Generics. However, that also means we only have access to the whole sequence. Creating references inside `clean_typing_generic_mro` might work though.